### PR TITLE
Add radare2

### PR DESCRIPTION
**Pull request by trufae**

### CODE REVIEW

File: bootstrap.yml [hunk 1]
```diff
@@ -514,6 +515,29 @@ packages:
...
           DESTDIR: '@THIS_COLLECT_DIR@'
 
+  - name: radare2
+    source:
+      subdir: '3rdparty'
+      url: 'https://codeload.github.com/radareorg/radare2/tar.gz/refs/tags/5.4.0'
+      format: 'tar.gz'
+      extract_path: 'radare2-5.4.0'
+      version: '5.4.0'
+    revision: 0
```
Comment (mintsuki) [inline bootstrap.yml:525]: Revisions start at 1, which is the default if this line is omitted. Usually this line is only added starting from `revision: 2`.

File: bootstrap.yml [hunk 2]
```diff
@@ -514,6 +515,29 @@ packages:
         environ:
           DESTDIR: '@THIS_COLLECT_DIR@'
 
+  - name: radare2
+    source:
+      subdir: '3rdparty'
+      url: 'https://codeload.github.com/radareorg/radare2/tar.gz/refs/tags/5.4.0'
+      format: 'tar.gz'
+      extract_path: 'radare2-5.4.0'
```
Comment (mintsuki) [inline bootstrap.yml:523]: Usually after this line goes `patch-path-strip: 3` which is needed for the patching script to do its magic. It is good use to add it even if the patch is empty, like for other packages.

File: bootstrap.yml [hunk 3]
```diff
@@ -514,6 +515,29 @@ packages:
...
+    revision: 0
+    tools_required:
+      - host-gcc
+    pkgs_required:
+      - mlibc
+    configure:
+      - args:
+        - 'bash'
+        - '-c'
+        - 'cd @THIS_SOURCE_DIR@ ; ./configure --without-debugger --host=x86_64-vinix --with-ostype=gnulinux --prefix=/usr'
```
Comment (mintsuki) [inline bootstrap.yml:534]: For packages that need in-tree builds, like `vim`, you should first copy the whole package source to `@THIS_BUILD_DIR@` first, then build it there. See the `vim` recipe.

**Review by mintsuki — CHANGES_REQUESTED** [review]

File: bootstrap.yml (hunk 1 above)
Comment (trufae) [inline bootstrap.yml:525]:
```suggestion
```

File: bootstrap.yml (hunk 3 above)
Comment (trufae) [reply bootstrap.yml]: like `cp -rf @THIS_SOURCE_DIR@/* .` ? i can also build with meson, but im afraid that will require some fixes to run in vinix and also will add meson+ninja+python as dependencies which is not what i want

Comment (mintsuki) [reply bootstrap.yml]: Yeah like that, just check the `vim` target.

**Comment by trufae** [comment]
Anything else to change? i cant get vinix to build so cant test locally :?

**Comment by Ccode-lang** [comment]
@trufae run in an action. I do this all the time for different reasons.

**Pull request merged**